test/ProblemDetailsResponseFactoryTest.php

```diff
@@ -176,4 +178,14 @@ public function testFactoryRendersPreviousExceptionsInDebugMode() : void
         $this->assertEquals(101010, $payload['exception']['stack'][0]['code']);
         $this->assertEquals('first', $payload['exception']['stack'][0]['message']);
     }
+
+    public function testFragileDataInExceptionMessageShouldBeHiddenInResponseBodyInNoneDebugMode()
+    {
+        $fragileMessage = 'Your SQL or password here';
+        $exception = new \Exception($fragileMessage);
+
+        $response = $this->factory->createResponseFromThrowable($this->request->reveal(), $exception);
+
+        $this->assertNotContains($fragileMessage, (string) $response->getBody());
```

Comment:
Okay, let me put it another way: if you make it empty and/or nullable, you'll have to change the signature of `createResponse()`, as it's currently a _required_ argument, or you'll have to change the internals of that method to test for an empty value and omit it from the payload.

That's a subject for another pull request entirely at that point: should `detail` be required or not?

Yes, the spec says it can be omitted. My argument is that `detail` is the one field that provides _human readable_ information that can help them understand what they need to correct, which is why I've made it a required argument.

In the case of exceptions, if we do not include the exception message, even something like "An error occurred on the server preventing processing of your request" is helpful, as then you know it's not likely something _you_ did, but a server-side issue or a bug in the API you're hitting.